### ISSUE TITLE
Creating complex navigation blocks

### ISSUE THREAD

**Issue by ghost** [ryanwelcher/twitch#21]
## Describe the issue your running into with your code
Can you make a custom block for megamenu's, off canvas menu's, etc. Basically how to make a custom navigation block.

**Comment by cthom-dev** [comment]
Yes, please Ryan can you?  I would move to FSE in a heartbeat if could code a custom navigation block, and mobile nav block (off canvas). Mega menu would be awesome. The native Nav block is way too restrictive. 

Something like what BricksBuilder has done with Nestable Nav, off Canvas and Mega menus. https://www.youtube.com/watch?v=krLjKKka9Xc&list=PLFYszb2Sm7DfXSATx-ccRuQRICaInwftK&index=12

**Comment by cthom-dev** [comment]
Hi @ryanwelcher , are you able to come back to this custom block and keep building it? I would really love to create a custom Nav block as the native one is very restrictive and limiting. Ideally a custom Nav Block and Custom Mobile Nav Block.
